python/ray/experimental/collective/reducescatter.py

```diff
@@ -0,0 +1,92 @@
+import logging
+from typing import List, Optional, Union
+
+import ray
+from ray.dag.collective_node import CollectiveOutputNode, _CollectiveOperation
+from ray.dag.constants import (
+    BIND_INDEX_KEY,
+    COLLECTIVE_OPERATION_KEY,
+    PARENT_CLASS_NODE_KEY,
+)
+from ray.experimental.channel.torch_tensor_type import Communicator, TorchTensorType
+from ray.experimental.util.types import ReduceOp
+from ray.util.collective.types import ReduceOp as RayReduceOp
+
+# TODO(wxdeng): Unify `ReduceOp` and `RayReduceOp`. Directly importing `RayReduceOp`
+# has dependency issues for some tests.
+
+logger = logging.getLogger(__name__)
+
+
+class ReduceScatterWrapper:
+    """Wrapper for NCCL reduce-scatter."""
+
+    def bind(
+        self,
+        input_nodes: List["ray.dag.DAGNode"],
+        op: ReduceOp = ReduceOp.SUM,
```

Comment:
After introducing the two new enums, this would be `ReduceScatterOp.SUM`. Also update for allreduce to use `AllReduceOp.SUM`. Make sure update all the places including `allreduce.py` and tests.